src/ResultResponseFactory/JsonResultResponseFactory.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace RstGroup\DiagnosticsMiddleware\ResultResponseFactory;
+
+use ArrayObject;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Diactoros\Response;
+use ZendDiagnostics\Result\Collection;
+use ZendDiagnostics\Result\FailureInterface;
+use ZendDiagnostics\Result\ResultInterface;
+use ZendDiagnostics\Result\SuccessInterface;
+use ZendDiagnostics\Result\WarningInterface;
+
+final class JsonResultResponseFactory implements ResultResponseFactoryInterface
+{
+    public function createResponse(ServerRequestInterface $request, Collection $resultCollection)
+    {
+        $data = [
+            'details' => $this->getDetails($resultCollection),
+            'success' => $resultCollection->getSuccessCount(),
+            'warning' => $resultCollection->getWarningCount(),
+            'failure' => $resultCollection->getFailureCount(),
+            'skip' => $resultCollection->getSkipCount(),
+            'unknown' => $resultCollection->getUnknownCount(),
+            'passed' => $resultCollection->getFailureCount() === 0,
+        ];
+
+        return new Response\JsonResponse($data);
+    }
+
+    private function getDetails(Collection $resultCollection)
```

Comment:
I dissagree. It's not container factory, but internal implementation of response factory. It's strictly connected with this what this class produce.